app/routes/populate/populate.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from "react"
+import { connect } from "react-redux"
+import NavFooter from "../shared/components/NavFooter"
+import PropTypes from "prop-types"
+import classNames from "classnames"
+import AssignmentCard from "./components/AssignmentCard"
+import LoadingPanel from "../shared/components/LoadingPanel"
+
+import {assignmentFetchInfo} from "../../modules/assignment/actions/assignment-fetch-info"
+import {setAssignmentURL} from "../../modules/assignment/actions/assignment-set-url"
+import {settingsLoginUser} from "../../modules/settings/actions/settings-login-user"
+import {url, error, valid, name, typeLabel, fetching} from "../../modules/assignment/selectors"
+import {userAuthorized} from "../../modules/settings/selectors"
+
+const containerStyles = {
+  paddingTop: "100px"
+}
+
+const cardStyles = {
+  marginTop: "75px"
+}
+
+const placeholderURL = "http://classroom.github.com/classrooms/your-org/assignments/your-assignment"
+
+class PopulatePage extends Component {
+  constructor (props) {
+    super(props)
+    this.updateInput = this.updateInput.bind(this)
+  }
+
+  componentDidMount () {
+    if (!this.props.loggedIn) {
```

Comment:
If user is not logged in when Assignment URL page is loaded, sends IPC message to open auth window and log in user.